pkg/client/types.go

```diff
@@ -75,3 +76,8 @@ func (c *immuClient) WithOptions(options *Options) *immuClient {
 	c.Options = options
 	return c
 }
+
+func (c *immuClient) WithErrorHandler(handler heartbeater.ErrorHandler) *immuClient {
```

Comment:
error handler should not be in package heartbeater, in the future more client errors can be handled and not just the ones related to heartbeats

Reply:
This is error handler for heartbeater exclusively, if defined on client level, leads to cyclic dependency so

Reply:
`func (c *immuClient) WithErrorHandler(handler ErrorHandler) `  client api should not expose internals. I think either the error handler is defined in the client package or `func (c *immuClient) WithHeartBeater(...)` is directly set as a component. But I'd think the first alternative is better as it gives the chance to handle more errors and not just those related to heartbeats...

Reply:
func (c *immuClient) WithErrorHandler(handler ErrorHandler)  client api should not expose internals. I think either the error handler is defined in the client package or func (c *immuClient) WithHeartBeater(...) is directly set as a component. But I'd think the first alternative is better as it gives the chance to handle more errors and not just those related to heartbeats...
regarding cyclic dependencies, probably it's not worth to have a hearbeater package..